habitat_baselines/rl/ddppo/algo/ddppo_trainer.py

```diff
@@ -0,0 +1,428 @@
+            env_time = requeue_stats["env_time"]
+            pth_time = requeue_stats["pth_time"]
+            count_steps = requeue_stats["count_steps"]
+            count_checkpoints = requeue_stats["count_checkpoints"]
+            start_update = requeue_stats["start_update"]
+            prev_time = requeue_stats["prev_time"]
+
+        with (
+            TensorboardWriter(
+                self.config.TENSORBOARD_DIR, flush_secs=self.flush_secs
+            )
+            if self.world_rank == 0
+            else contextlib.suppress()
+        ) as writer:
+            for update in range(start_update, self.config.NUM_UPDATES):
+                if ppo_cfg.use_linear_lr_decay:
+                    lr_scheduler.step()
+
+                if ppo_cfg.use_linear_clip_decay:
+                    self.agent.clip_param = ppo_cfg.clip_param * linear_decay(
+                        update, self.config.NUM_UPDATES
+                    )
+
+                requeue_stats = dict(
+                    env_time=env_time,
+                    pth_time=pth_time,
+                    count_steps=count_steps,
+                    count_checkpoints=count_checkpoints,
+                    start_update=update,
+                    prev_time=(time.time() - t_start) + prev_time,
+                )
+                count_steps_delta = 0
+                self.agent.eval()
+                for step in range(ppo_cfg.num_steps):
+                    if EXIT.is_set():
+                        self.envs.close()
+
+                        if REQUEUE.is_set() and self.world_rank == 0:
+                            save_interrupted_state(
+                                dict(
+                                    state_dict=self.agent.state_dict(),
+                                    optim_state=self.agent.optimizer.state_dict(),
+                                    lr_sched_state=lr_scheduler.state_dict(),
+                                    config=self.config,
+                                    requeue_stats=requeue_stats,
+                                )
+                            )
+
+                        requeue_job()
+                        return
+
+                    (
+                        delta_pth_time,
+                        delta_env_time,
+                        delta_steps,
+                    ) = self._collect_rollout_step(
+                        rollouts,
+                        current_episode_reward,
+                        episode_rewards,
+                        episode_counts,
+                    )
+                    pth_time += delta_pth_time
+                    env_time += delta_env_time
+                    count_steps_delta += delta_steps
+
+                    # This is where the preemption of workers happens.  If a
+                    # worker detects it will be a straggler, it preempts itself!
+                    if (step >= ppo_cfg.num_steps / 4) and int(
```

Comment:
Thank you for adding the comment. For understanding that would be good to describe motivation behind `(step >= ppo_cfg.num_steps / 4)` and maybe make a constant `0.25` with a reasonable name.

Reply:
Maybe we can name it as `SHORT_ROLLOUT_THRESHOLD` and add comment near definition, regarding why short rollouts hurt.